test/functional/pdo_sqlsrv/MsCommon_mid-refactor.inc

```diff
@@ -147,11 +182,11 @@ class ColumnMeta
      */
     public function getColDef()
     {
-        //return getColDef($this->colName, $this->dataType, $this->options, $this->encType);
         $append = " ";
 
         // an identity column is not encrypted because a select query with identity column as the where clause is often run and the user want to have to bind parameter every time
-        if (isColEncrypted() && $this->isEncryptableType() && stripos($this->options, "identity") === false && stripos($this->options, "rowguidcol") === false) {
+        if (isColEncrypted() && ($this->encType == "deterministic" || $this->encType == "ramdomized") && $this->isEncryptableType() 
+            && stripos($this->options, "identity") === false && stripos($this->options, "rowguidcol") === false) {
             $cekName = getCekName();
```

Comment:
why don't you add the check for identity and rowguidcol to isEncryptableType()? (might consider changing the name to isEncryptable()...)